src/hydrate.ts

```diff
@@ -18,6 +25,32 @@ type ParseModeFlavor<C extends Context> = C & {
   replyWithMarkdown: C["reply"];
   replyWithMarkdownV1: C["reply"];
   replyWithMarkdownV2: C["reply"];
+
+  replyFmtWithPhoto: (
+    photo: string,
+    options: {
+      caption: Stringable;
+      reply_markup?: Parameters<C["replyWithPhoto"]>[1]["reply_markup"];
+    }
+  ) => ReturnType<C["replyWithPhoto"]>;
+
+  replyFmtWithMediaGroup: (
+    media: (InputMediaPhoto | InputMediaVideo | InputMediaAnimation | InputMediaAudio | InputMediaDocument & { caption: Stringable })[]
```

Comment:
The `caption` is always optional. We should not make it required here.

The manual union is workable. Pulling the type from `Parameters<C[themethodname]>` would be superior.